src/main/java/com/exasol/adapter/request/parser/TablesMetadataParser.java

```diff
@@ -0,0 +1,169 @@
+package com.exasol.adapter.request.parser;
+
+import com.exasol.adapter.metadata.ColumnMetadata;
+import com.exasol.adapter.metadata.DataType;
+import com.exasol.adapter.metadata.MetadataException;
+import com.exasol.adapter.metadata.TableMetadata;
+
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonString;
+import javax.json.JsonValue;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TablesMetadataParser {
+    private static final String INVOLVED_TABLES = "involvedTables";
+    private static final String DATA_TYPE = "dataType";
+
+    public List<TableMetadata> parse(final JsonObject tablesAsJson) throws MetadataException {
+        final List<TableMetadata> tables = new ArrayList<>();
+        final JsonArray jsonArray = tablesAsJson.getJsonArray(INVOLVED_TABLES);
+        for (final JsonObject table : jsonArray.getValuesAs(JsonObject.class)) {
+            final String tableName = table.getString("name", "");
+            final String tableAdapterNotes = readAdapterNotes(table);
+            final String tableComment = table.getString("comment", "");
+            final List<ColumnMetadata> columns = new ArrayList<>();
+            for (final JsonObject column : table.getJsonArray("columns").getValuesAs(JsonObject.class)) {
+                columns.add(parseColumnMetadata(column));
+            }
+            tables.add(new TableMetadata(tableName, tableAdapterNotes, columns, tableComment));
+        }
+        return tables;
+    }
+
+    private ColumnMetadata parseColumnMetadata(final JsonObject column) {
+        final String columnName = column.getString("name");
+        final String adapterNotes = readAdapterNotes(column);
+        final String comment = column.getString("comment", "");
+        final String defaultValue = column.getString("default", "");
+        final boolean isNullable = checkIfBooleanValueExists(column, "isNullable");
```

Comment:
The name of the method is of. You are not checking the existence as much as you are applying a default value if it does not exist.

Reply:
fixed